instrumentation/spring/spring-boot-autoconfigure/src/test/java/io/opentelemetry/instrumentation/spring/autoconfigure/OpenTelemetryAutoConfigurationTest.java

```diff
@@ -58,46 +57,28 @@ void customOpenTelemetry() {
   void initializeProvidersAndOpenTelemetry() {
     this.contextRunner
         .withConfiguration(AutoConfigurations.of(OpenTelemetryAutoConfiguration.class))
-        .run(
-            context ->
-                assertThat(context)
-                    .hasBean("openTelemetry")
-                    .hasBean("sdkTracerProvider")
-                    .hasBean("sdkMeterProvider")
-                    .hasBean("sdkLoggerProvider"));
+        .run(context -> assertThat(context).hasBean("openTelemetry"));
   }
 
   @Test
   @DisplayName(
-      "when Application Context DOES NOT contain OpenTelemetry bean but TracerProvider should initialize openTelemetry")
+      "when Application Context DOES NOT contain OpenTelemetry bean but SpanExporter should initialize openTelemetry")
   void initializeOpenTelemetryWithCustomProviders() {
+    OtlpSpanExporterProvider customSpanExporter =
+        Mockito.mock(
+            OtlpSpanExporterProvider.class,
+            withSettings().extraInterfaces(AutoConfigureListener.class));
+
     this.contextRunner
         .withBean(
-            "customTracerProvider",
-            SdkTracerProvider.class,
-            () -> SdkTracerProvider.builder().build(),
-            bd -> bd.setDestroyMethodName(""))
-        .withBean(
-            "customMeterProvider",
-            SdkMeterProvider.class,
-            () -> SdkMeterProvider.builder().build(),
-            bd -> bd.setDestroyMethodName(""))
-        .withBean(
-            "customLoggerProvider",
-            SdkLoggerProvider.class,
-            () -> SdkLoggerProvider.builder().build(),
+            "customSpanExporter",
+            OtlpSpanExporterProvider.class,
+            () -> customSpanExporter,
             bd -> bd.setDestroyMethodName(""))
         .withConfiguration(AutoConfigurations.of(OpenTelemetryAutoConfiguration.class))
-        .run(
-            context ->
-                assertThat(context)
-                    .hasBean("openTelemetry")
-                    .hasBean("customTracerProvider")
-                    .doesNotHaveBean("sdkTracerProvider")
-                    .hasBean("customMeterProvider")
-                    .doesNotHaveBean("sdkMeterProvider")
-                    .hasBean("customLoggerProvider")
-                    .doesNotHaveBean("sdkLoggerProvider"));
+        .run(context -> assertThat(context).hasBean("openTelemetry"));
+
+    Mockito.verify(customSpanExporter).afterAutoConfigure(any());
```

Comment:
this is the test that makes sure that the spring bean gets picked up by sdk auto config